### PR TITLE
G Suite: Display discount in shopping cart

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request displays and formats the price in the shopping cart when a G Suite discount is applied. It also displays a short text next to the price:

Popup | Checkout 
------ | -----
 | screenshot

Finally, it introduces constants to avoid referring to slugs directly, and to take ease future refactoring.

#### Testing instructions

1. Apply 2332e-pb to your sandbox to simulate a discount to the frontend
2. Apply D36169-code to to your sandbox to help differentiate between different discounts.
3. Point `public-api.wordpress.com` to your sandbox
4. Run `git checkout add/gsuite-discount-shopping-cart` and start your development server
5. Log in to an account that has a site without G Suite
6. Open the [`Email` page](http://calypso.localhost:3000/email), then click the `Add G Suite` button
7. Fill in the form, and click the `Continue` button
8. Assert that the shopping cart looks like the screenshot above
9. Navigate back to the [`Plans`](http://calypso.localhost:3000/plans)
10. Click the `Cart` link in the upper right corner of the cake header
11. Assert that the shopping cart looks like the screenshot above

/cc @fditrapani

### CODE REVIEW

File: client/my-sites/checkout/cart/cart-item.jsx [hunk 1]
```diff
@@ -63,6 +65,28 @@ export class CartItem extends React.Component {
...
+						<span className="cart__gsuite-discount-regular-price">
+							{ costBeforeCoupon }
+						</span>
+
+						<span className="cart__gsuite-discount-discounted-price">
+							{ cost } { cartItem.currency }
+						</span>
+
+						<span className="cart__gsuite-discount-text">
+							{ translate( 'First year discount' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/cart/cart-item.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Discount for first year' )` *ES Score: 9*

**Comment by stephanethomas** [comment]
Some linting tests are failing for existing code. I don't plan to fix them as this would make this pull request way too long.

File: client/my-sites/checkout/cart/cart-item.jsx (hunk 1 above)
Comment (stephanethomas) [reply client/my-sites/checkout/cart/cart-item.jsx]: I suppose that's a good option, bot. @fditrapani, let me know what you think of the copy 🙂.

Comment (fditrapani) [reply client/my-sites/checkout/cart/cart-item.jsx]: The copy looks good to me 👍

**Comment by stephanethomas** [comment]
I've rebased this pull request, and updated the [copy of the discount](https://github.com/Automattic/wp-calypso/pull/37978#discussion_r350866293).

File: client/my-sites/checkout/cart/cart-item.jsx [hunk 2]
```diff
@@ -63,6 +65,28 @@ export class CartItem extends React.Component {
 			return <span className="cart__free-text">{ translate( 'Free' ) }</span>;
 		}
 
+		if ( isGSuiteProductSlug( cartItem.product_slug ) ) {
+			const { cost_before_coupon: costBeforeCoupon } = cartItem;
+
+			if ( costBeforeCoupon ) {
```
Comment (stephanethomas) [inline client/my-sites/checkout/cart/cart-item.jsx]: We should revisit that condition. I've indeed encountered the case where the API would return a value for this property even if no coupon was applied.

Comment (stephanethomas) [reply client/my-sites/checkout/cart/cart-item.jsx]: This also doesn't handle very well the case where the user enters a coupon.

Comment (stephanethomas) [reply client/my-sites/checkout/cart/cart-item.jsx]: I think I found a way to address my first comment. I'm still looking how to handle multiple coupons.

Comment (olaseni) [reply client/my-sites/checkout/cart/cart-item.jsx]: This should be fixed with the `is_sale_coupon_applied` property in D36169-code. It's more reliable to depend on that.

File: client/my-sites/checkout/cart/cart-item.jsx [hunk 3]
```diff
@@ -66,21 +66,25 @@ export class CartItem extends React.Component {
...
+						<span className="cart__gsuite-discount-regular-price">{ costBeforeCoupon }</span>
 
 						<span className="cart__gsuite-discount-discounted-price">
 							{ cost } { cartItem.currency }
 						</span>
 
 						<span className="cart__gsuite-discount-text">
-							{ translate( 'Discount for first year' ) }
+							{ isCouponApplied
+								? translate( 'Multiple discounts' )
```
Comment (stephanethomas) [inline client/my-sites/checkout/cart/cart-item.jsx]: I think we could also simply display nothing when both a coupon code (i.e. a coupon entered by the user on the checkout page) as well as a sale coupon (i.e. a coupon automatically applied to a product such as the two-month discount) apply. @fditrapani, what do you think?

Comment (fditrapani) [reply client/my-sites/checkout/cart/cart-item.jsx]: It would still be helpful to show something. "Discounts applied"  would be a better way to word it.

Comment (stephanethomas) [reply client/my-sites/checkout/cart/cart-item.jsx]: @fditrapani: we don't display anything when the user enters a coupon code for other products (actually we don't even cross the corresponding costs out). Do you still think we should display `Discounts applied` even if it's not consistent?

No coupon code | With coupon code
------ | -----
screenshot | screenshot

File: client/my-sites/checkout/cart/style.scss [hunk 4]
```diff
@@ -69,6 +69,25 @@
 			padding-left: 0;
 		}
 
+		.cart__gsuite-discount {
+			.cart__gsuite-discount-regular-price {
+				text-decoration: line-through;
+
+				&:empty + .cart__gsuite-discount-discounted-price {
+					padding-left: 0;
+				}
```
[the remark this answers is not in the document]
Comment (olaseni) [reply client/my-sites/checkout/cart/style.scss]: Oh that was for the situation when the coupon is no more, while the discount had previously been applied. That shouldn't happen anymore. Will fix it.

[the remark this answers is not in the document]
Comment (olaseni) [reply client/my-sites/checkout/cart/style.scss]: Fixed.

**Comment by fditrapani** [comment]
> @fditrapani: we don't display anything when the user enters a coupon code for other products (actually we don't even cross the corresponding costs out). Do you still think we should display Discounts applied even if it's not consistent?

Yeah, I think it's fine. We will deal with this more consistently in the new signup.

**Pull request merged**